### ISSUE TITLE
Implement PagerDuty

### ISSUE THREAD

**Issue by bensternthal** [mozmeao/infra#692]
This issue tracks the implementation, documentation and training on pagerduty.

- [x] Get pager duty accounts for everyone that needs them.
- [x] Propose pagerduty <--> new relic integration to SRE team.
- [x] Document escalation path & pager duty integration on mana.
- [x] Create escalation paths for each app in PD
- [x] Create new relic alerts for each app.
- [x] Link pagerduty & new ....new relic integration.
- [x] Train team and create documentation,
- [x] Add email filter to PG to alert only on specific emails.

**Comment by bensternthal** [comment]
Draft plan for implementing pagerduty.

### Current Setup
New relic currently has 2 production alert policies.

1. Default Application Alert Policy - <0.7 apdex | >5% Error | Ping fails for > 5 minutes.
2. MDN Alerts -  <0.7 apdex | >5% Error | Ping fails for > 5 minutes.

Both send an email to our meao-alerts slack channel.

### Proposed Setup

Since we are favoring  static escalation paths per application rather than on overall oncall program we will need to configure new relic to _ping_ the right application in pagerduty rather than just sending an email or api call to a generic endpoint.

### Proposed Changes
To accomplish the above we will need to:

#### Pager Duty
1. Create _services_ for each of our applications with a unique email address **service.mozmeao@mozilla.pagerduty.com**

#### New Relic

1. Create an application policy for any application with a distinct escalation path.
2. Configure the policy to ping both the slack channel and the unique pager duty email.

**Comment by metadave** [comment]
Looks good so far.

**Comment by bensternthal** [comment]
OK I am moving on this again :)

Below is a draft escalation path. We will have one of these for each site. The logic I am working under is:

1. Whomever is listed as primary [here](https://mana.mozilla.org/wiki/pages/viewpage.action?pageId=50267455) is contacted first.
2. And as you might suspect whomever is secondary is.... contacted second.
3. Third contact will likely always be an SRE.
4 Fourth contact can be me. I can be the human of last resort.
5. I am thinking 15 minutes as the delay between trying to escalate to the next person.

Attached is a screenshot for the escalation path for Moz.org.

@jgmize @metadave let me know if the above is good to go and I will work on for all the services.


<img width="1097" alt="screen shot 2018-01-19 at 8 49 26 am" src="https://user-images.githubusercontent.com/198507/35161661-b4a2fe96-fcf5-11e7-9444-cf6a91657f2c.png">

**Comment by metadave** [comment]
LGTM

**Comment by bensternthal** [comment]
@metadave @glogiotatidis @jwhitlock @jgmize @pmac @escattone 

Folks, we are getting close to enabling pagerduty. In order to do so I need each of you to do the following.

1. Please review [this mana page](https://mana.mozilla.org/wiki/pages/viewpage.action?pageId=50267455) and make sure you are comfortable with the applications you are listed as a contact for.
2. Please login to pagerduty and adjust your contact information and your notification rules.
3. Review the [incident response guide](https://github.com/mozmeao/infra/blob/master/docs/incident_response.md), if you are responding to an incident these are the steps to follow.
4. If you have questions, add them here... we can both answer them and add the answers to our documentation!

To ensure that I do not assign folks to escalation paths without having things setup properly, once you complete the 3 above steps please check off your name:

- [ ] Ben Sternthal 
- [ ] Dave Parfitt
- [ ] Giorgos Logiotatidis 
- [ ] John Whitlock
- [ ] Josh Mize
- [ ] Paul McLanahan
- [ ] Ryan Johnson

Once the above is done I will connect new relic and pager duty and we can start using the new system.

**Comment by jwhitlock** [comment]
This isn't a question, more of a TODO for me in a new PR.

MDN is going through some growing pains in AWS. It is handling normal traffic, but alerts are raised for certain scraper behavior. We're working on long-term solutions, such as a CDN, but the short-term process is to figure out the scraper IP and block it. MDN will need a living document for alert handling, including setup steps that an incident responder will want to do before handling a 3 AM alert.  Until then, the next step will be "wake up John" 😄

**Comment by bensternthal** [comment]
Issues I found:

* Bedrock Ping URL Only Set For US est
* No alert policy for IRL podcast.
* Nucleus "Sugar" - Unsure if this is production app or something else.
* Nucleus Does Not Have Ping URLs Set
* Basket Ping URL Only Has Admin Us West Set
* Snippets Only Has _admin oregon b_ and _prod oregon-b_ in an alert policy
* Snippets Has No Ping URL Set
* MDN Alerts Only Has Kuma-Web-Prod In alert policy
